File: Gixor-Backend/Gixor-Api/Gixor-Api-GitHub/src/main/java/com/cpy3f2/Gixor/Domain/Event.java
package com.cpy3f2.Gixor.Domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 *  GitHub Event
 * @author simon
 * @since 2024/11/5 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class Event {
    private String id;
    private String type;
    private Actor actor;
    private Repo repo;
    private Payload payload;
    private Boolean isPublic;
    
    @JsonProperty("created_at")
    private LocalDateTime createdAt;

    @Data
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class Actor {
        private Long id;
        private String login;
        @JsonProperty("display_login")
        private String displayLogin;
        @JsonProperty("gravatar_id")
        private String gravatarId;
        private String url;
        @JsonProperty("avatar_url")
        private String avatarUrl;
    }

    @Data
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class Repo {
        private Long id;
        private String name;
        private String url;
    }

    @Data
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class Payload {
        private String action;
        @JsonProperty("push_id")
        private Long pushId;
        private Integer size;
        @JsonProperty("distinct_size") 
        private Integer distinctSize;
        private String ref;
        private String head;
        private String before;
        private String description;
        private String masterBranch;
    }
} 

File: Gixor-Mobile/app/src/main/java/com/cpy3f2/gixor_mobile/model/entity/Event.kt
package com.cpy3f2.gixor_mobile.model.entity
import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

/**
 * GitHub Event
 * @author lya
 * @since 2024/11/5
 */
data class Event(
    val id: String,
    val type: String,
    val actor: Actor,
    val repo: Repo,
    val payload: Payload,
    val isPublic: Boolean,

    @SerializedName("created_at")
    val createdAt: String
) {

    data class Actor(
        val id: Long,
        val login: String,
        @SerializedName("display_login")
        val displayLogin: String,
        @SerializedName("gravatar_id")
        val gravatarId: String,
        val url: String,
        @SerializedName("avatar_url")
        val avatarUrl: String
    )

    data class Repo(
        val id: Long,
        val name: String,
        val url: String
    )

    data class Payload(
        val action: String,
        @SerializedName("push_id")
        val pushId: Long,
        val size: Int,
        @SerializedName("distinct_size")
        val distinctSize: Int,
        val ref: String,
        val head: String,
        val before: String,
        val description: String,
        val masterBranch: String
    )
}
